src/main/java/com/flansmod/common/guns/EntityGrenade.java

```diff
@@ -554,7 +614,8 @@ protected void writeEntityToNBT(NBTTagCompound tags)
 	public void writeSpawnData(ByteBuf data)
 	{
 		ByteBufUtils.writeUTF8String(data, type.shortName);
-		data.writeInt(thrower == null ? 0 : thrower.getEntityId());
+		data.writeInt(player.isPresent() ? player.get().getEntityId() : -1);
+		data.writeInt(thrower.isPresent() ? thrower.get().getEntityId() : -1);
```

Comment:
Previous behaviour was to set ID to 0, -1 makes more sense as an ID that won't exist but will this break anything?

Reply:
It should not break something. The `getEntityByID` just returns null if it cannot find that entity and '-1' is a perfectly fine integer. I also tested the code with these parameters and could not observe any problems